server/aesdsocket.c

```diff
@@ -0,0 +1,429 @@
+        {
+            printf("Connection accepted\n");
+            syslog(LOG_ERR, "New Connection accepted");
+        }
+
+        //Printing IP address of client
+        char address[30];   //To store the IP address converted into string
+        
+        //const char *inet_ntop(int af, const void *restrict src,
+                          //   char *restrict dst, socklen_t size);
+        const char*ip_addr = inet_ntop(AF_INET, &client.sin_addr, address, sizeof(address));
+        int client_port = htons(client.sin_port);
+        if(ip_addr == NULL)
+        {
+            printf("Error in obtaining ip address of the client\r\n");
+        }
+        else
+        {
+            printf("Accepted connection from %s %d\r\n", ip_addr,client_port);
+            //Log this info
+            syslog(LOG_INFO,"Accepted connection from %s %d\r\n", ip_addr,client_port);
+        }
+
+	//Initial malloc
+	    char *buffer = (char*) malloc(MAX_SIZE);
+	    len_tracker = MAX_SIZE;
+	    if(buffer == NULL)
+	    {
+           	printf("Error in Malloc\r\n");
+            	//Log this info
+            	syslog(LOG_INFO,"Error in Malloc\r\n");
+	    	return -1;
+	    }
+
+	    //memset the buffer so it has all zeros
+	     memset(buffer,0,MAX_SIZE);
+
+        //Reception and Transmission
+        while(1)
+        {
+	    
+	    //reads from the client, store it in a buffer
+            buffer_len = read(cli_fd, buffer + len_tracker - MAX_SIZE, MAX_SIZE);
+		
+            if(buffer_len == 0)
+            {
+                printf("Disconnected from client\n");
+                break;
+            }
+	    if(buffer_len < 0)
+	    {
+	    	printf("Error in reading from the client\r\n");
+		break;
+	    }
+	
+            //stop with newline character
+            for(index = 0; index < buffer_len; index++)
+            {
+                if (buffer[index + len_tracker - MAX_SIZE] == '\n')
+                {
+		    break;
+                }
+            }
+            
+
+	   //if(buffer_len < MAX_SIZE)
+	    if(index < buffer_len)
+	   {	  
+	   	printf("2\n"); 	
+		//can write to file directly
+		//fwrite(buffer, buffer_len, 1, fp);
+		if(fwrite(buffer, len_tracker - MAX_SIZE + index + 1, 1, fp) == -1)
+		 {
+           		printf("Error in writing to the file\r\n");
+           		//Log this info
+           		syslog(LOG_INFO,"Error in writing to the file. Error: %d\r\n", errno);
+    	    		free(buffer);
+			buffer = NULL;
+    	    		graceful_exit();
+           		 return -1;
+      		  }
+      		  break;
+		
+	   }
+	   else
+	   {
+   		printf("3\n");
+           	len_tracker += MAX_SIZE;
+           	buffer = (char*)realloc(buffer,len_tracker);
+		  
+	   }
+	}
+	printf("1\n");
+	len_tracker = 0;
+	free(buffer);
+	buffer = NULL;
+	int file_bytes = 0;
+        char ch;
+        fseek(fp, 0, SEEK_SET);
+        while(fread(&ch, 1, 1, fp) > 0)
+        {
+            file_bytes++;
+        }
+        char *write_buffer = (char*) malloc (file_bytes);
```

Comment:
@adna4700 what assumption are you making here about the size of the process heap and the total size of all packets sent?